packages/algob/src/lib/script-checkpoints.ts

```diff
@@ -81,23 +82,43 @@ export class CheckpointRepoImpl implements CheckpointRepo {
   strippedCP: Checkpoints = {};
   precedingCP: Checkpoints = {};
   allCPs: Checkpoints = {};
+  scriptMap: assetScriptMap = {};
 
-  private _mergeTo (target: Checkpoints, cp: Checkpoints): Checkpoints {
+  private _mergeTo (target: Checkpoints, cp: Checkpoints, scriptMap: assetScriptMap): Checkpoints {
     const keys: string[] = Object.keys(cp);
     return keys.reduce((out: Checkpoints, key: string) => {
       return appendToCheckpoint(out, key, cp[key]);
     }, target);
   }
 
-  merge (cp: Checkpoints): CheckpointRepo {
+  merge (cp: Checkpoints, scriptName: string): CheckpointRepo {
     this.strippedCP = cp;
-    this.precedingCP = this._mergeTo(this.precedingCP, cp);
-    this.mergeToGlobal(cp);
+    this.precedingCP = this._mergeTo(this.precedingCP, cp, this.scriptMap);
+    this.mergeToGlobal(cp, scriptName);
     return this;
   }
 
-  mergeToGlobal (cp: Checkpoints): CheckpointRepo {
-    this.allCPs = this._mergeTo(this.allCPs, cp);
+  mergeToGlobal (cp: Checkpoints, scriptName: string): CheckpointRepo {
+    this.allCPs = this._mergeTo(this.allCPs, cp, this.scriptMap);
+
+    if (scriptName.charAt(0) === 'a') {
+      scriptName = scriptName.slice(10, scriptName.length - 8);
```

Comment:
You either have to pass script names into this method or checkpoint names. Not both. You don't need this check.

`10` is hardcoded. 
`=== 'a'` is only the first letter, it will be true for `a_scripts1` too which shouldn't be matched.
8 is hardcoded. It will crash if we'll change our checkpoint ending from `.cp.yaml` to anything that is not of length 8.

Extract this into a method.

Reply:
will change the input from parent methods

Reply:
I have this `.cp.yaml` saved somewhere in the code. Or it should be extracted and ending should be checked.
`string.endsWith(...)`